File: 03. JavaScriptBasics-Homeworks/02. JavaScriptSyntax/04. CheckIfNumberIsPrime/primeChecker.js
'use strict';

function isPrime(number) {

    if (number != 0) {

        if (number == 1) {
            return false;
        }

        else {

            var prime = true;

            for (var i = 2; i < number/2 ; i++) {

                if (number % i == 0) {
                    prime = false;
                    break;
                }

            }

            return prime;
        }
    }
}

console.log(isPrime(7));
console.log(isPrime(254));
console.log(isPrime(587));

File: 00. CSharpBasics-Homeworks/07. CSharpAdvancedTopics/02.PrimeChecker/PrimeChecker.cs
using System;

class PrimeChecker
{
    static void Main(string[] args)
    {
        Console.Write("Enter number: ");
        ulong n = ulong.Parse(Console.ReadLine());

        bool Prime = CheckPrime(n);

        Console.WriteLine("Is Prime: {0}",Prime);
    }

    private static bool CheckPrime(ulong x)
    {
        bool isPrime = true;

        if (x == 0 || x == 1)
        {
            isPrime = false;
            return isPrime;
        }
        else
        {
            for (ulong i = 2; i <= Math.Sqrt(x); i++)
            {
                if (x % i == 0)
                {
                    isPrime = false;
                    break;
                }
            }
        }

        return isPrime;
    }
}

